broker/src/crypto.rs

```diff
@@ -137,22 +129,20 @@ impl GetCertsFromPki {
         method: &Method,
         api_path: &str,
         max_tries: Option<u32>,
-    ) -> Result<Response<Body>, SamplyBeamError> {
-        debug!("Samply.PKI: Vault request to {api_path}");
+    ) -> Result<reqwest::Response, SamplyBeamError> {
         let uri = pki_url_builder(api_path);
+        debug!("Samply.PKI: Vault request to {uri}");
         let max_tries = max_tries.unwrap_or(u32::MAX);
         for tries in 0..max_tries {
             if tries > 0 {
                 tokio::time::sleep(Duration::from_secs(3)).await;
             }
-            let req = Request::builder()
-                .method(method)
+            let resp = self.hyper_client
+                .request(method.clone(), uri.clone())
```

Comment:
1. We take method by ref to this fn so that would need to change.
2. That still would not prevent the clone because we retry in a loop so we cant move it.